python/lsst/summit/utils/peekExposure.py

```diff
@@ -968,10 +970,16 @@ def runPeek(
             # donut mode.
             if donutDiameter > self.config.donutThreshold:
                 mode = "donut"
-            elif isDispersedExp(exposure):
-                mode = "spec"
             else:
-                mode = "photo"
+                try:
+                    isDisp = isDispersedExp(exposure)
+                except Exception as e:
+                    self.log.warning(f"isDispersedExp failed: {e}.  Assuming not dispersed")
+                    isDisp = False
+                if isDisp:
+                    mode = "spec"
+                else:
+                    mode = "photo"
```

Comment:
There's two things I don't love here: exceptions as control flow (not so bad given this won't be in a tight loop), and the fact this will routinely spew really quite scary looking warnings (made somewhat worse still because it includes the whole "{e}" in the message) during normal/expected operation (which is really bad behaviour IMO). I think it shouldn't be too hard to avoid that though. I've just pushed a commit (untested) that I think should give you what you want though. Could you test that it does what you want? If it does, then that's a 👍 from me (obviously).

Reply:
Thanks, that looks good to me.  I've reordered some of the if/else logic and removed the unnecessary `isDisp` variable, but followed your same logic.